packages/calypso-e2e/src/lib/components/cookie-banner-component.ts

```diff
@@ -32,6 +32,11 @@ export class CookieBannerComponent {
 
 		// Whether the cookie banner appears is not deterministic.
 		// If it is not present, exit early.
+		try {
+			await locator.waitFor( { timeout: 100 } );
```

Comment:
I wonder if this would work:

```
if ( !( await locator.isVisible() ) ) {
  return;
}
```

Though it might not be feasible depending on how the cookie banner is rendered (i.e if it's not in the DOM at the time this is called).

Reply:
https://playwright.dev/docs/api/class-locator#locator-is-visible says about its `timeout` option

> Deprecated This option is ignored. locator.isVisible() does not wait for the element to become visible and returns immediately

Which makes me think it won't fix the slight delay problem.

Reply:
I think the alternate (`locator.toBeVisible( { timeout: 1000 } )` could work, allowing it to continue as soon as it's visible

Reply:
Because the `click` action waits for the element to be visible by default, I think we can flatten the whole thing to just:

```js
try {
  locator.click( { timeout: 1000 } );
catch ( _error ) {
 // Cookie banner was not displayed, so let's move on.
}
```

It's still not very resilient, though, as we're not really able to nail the perfect timeout value here. On the other hand, if we use a _safe_ timeout value (idk, 10s? 🤷) we'll be delaying each test by 10 seconds for no reason. 😅 

I guess it should be treated as a special kind of check that is run only after the page has fully loaded, so there's no need to do any amount of waiting (1s is enough). Having said that, I'm not sure what "page fully loaded" means if we have non-deterministic elements on it! 😬

Reply:
Looking at the selectors, this doesn't seem to be the Calypso cookie banner, but rather the generic one. There are a bunch of conditions, like if the user already consented, or if the user is not from a GDPR country, etc. This README is a good starting point for all the conditions with additional materials to understand it (it's a pretty complex bit): fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sn8p%2Qnanylgvpf%2S-og